paparazzi/paparazzi-gradle-plugin/src/main/java/app/cash/paparazzi/gradle/PaparazziPlugin.kt

```diff
@@ -104,21 +107,43 @@ class PaparazziPlugin : Plugin<Project> {
         val nonTransitiveRClassEnabled =
           (project.findProperty("android.nonTransitiveRClass") as? String).toBoolean()
 
+        // local resources
+        val localResourceFiles = project
+          .files(variant.sourceSets.flatMap { it.resDirectories })
+          .asFileTree
+
+        // library resources
+        // https://android.googlesource.com/platform/tools/base/+/96015063acd3455a76cdf1cc71b23b0828c0907f/build-system/gradle-core/src/main/java/com/android/build/gradle/tasks/MergeResources.kt#875
+        val libraryResourceFiles = variant.runtimeConfiguration
+          .incoming
+          .artifactView { config: ArtifactView.ViewConfiguration ->
+            config.attributes { container: AttributeContainer ->
+              container.attribute(
+                AndroidArtifacts.ARTIFACT_TYPE,
+                AndroidArtifacts.ArtifactType.ANDROID_RES.type
+              )
+            }
+          }
+          .artifacts
+          .artifactFiles
+          .asFileTree
+
         task.packageName.set(android.packageName())
         task.artifactFiles.from(packageAwareArtifacts.artifactFiles)
         task.nonTransitiveRClassEnabled.set(nonTransitiveRClassEnabled)
         task.mergeResourcesOutputDir.set(buildDirectory.asRelativePathString(mergeResourcesOutputDir))
         task.targetSdkVersion.set(android.targetSdkVersion())
         task.compileSdkVersion.set(android.compileSdkVersion())
         task.mergeAssetsOutputDir.set(buildDirectory.asRelativePathString(mergeAssetsOutputDir))
+        task.localResourceFiles.from(localResourceFiles)
+        task.libraryResourceFiles.from(libraryResourceFiles)
         task.paparazziResources.set(buildDirectory.file("intermediates/paparazzi/${variant.name}/resources.txt"))
       }
 
       val testVariantSlug = testVariant.name.capitalize(Locale.US)
 
       project.plugins.withType(JavaBasePlugin::class.java) {
-        project.tasks.named("compile${testVariantSlug}JavaWithJavac")
-          .configure { it.dependsOn(writeResourcesTask) }
+        writeResourcesTask.configure { it.dependsOn(project.tasks.named("compile${testVariantSlug}JavaWithJavac")) }
```

Comment:
since we're passing FileCollections now (previously we were eagerly dereferencing their contents!), we shouldn't need any of these task rewiring anymore.

so let's remove!